src/features/auth/authSaga.js

```diff
@@ -1,50 +1,126 @@
-import { put, call, takeEvery, all } from "redux-saga/effects";
-import axios from "axios";
+import { put, call, takeLatest, all, fork } from "redux-saga/effects";
 
-import { login } from "./authSlice";
+import { authenication } from "../../configs/firebase";
+import {
+  signOut,
+  signInWithPopup,
+  GoogleAuthProvider,
+  createUserWithEmailAndPassword,
+  signInWithEmailAndPassword,
+  updateProfile,
+} from "firebase/auth";
 
-// import { authenication } from "../../configs/firebase";
-import { signInWithPopup, GoogleAuthProvider } from "firebase/auth";
+import {
+  signupRequest,
+  loginRequest,
+  loginSuccess,
+  loginFailure,
+  logoutRequest,
+  logoutSuccess,
+  logoutFailure,
+} from "./authSlice";
+import userApi from "../../utils/api/user";
 
-// 예시 추후엔 별도의 모듈로 관리...
-const loginApi = ({ token }) => {
-  axios.post("/login", { token: token }, { withCredentials: true });
-};
+function* userLogin({ payload }) {
+  try {
+    if (payload) {
+      const { email, password } = payload;
+      const response = yield signInWithEmailAndPassword(
+        authenication,
+        email,
+        password
+      );
+      const { accessToken: token } = response.user;
 
-function* userLogin() {
-  const provider = new GoogleAuthProvider();
+      const apiResult = yield call(userApi.getlogin, token);
```

Comment:
```suggestion
      const response = yield call(userApi.getlogin, token);
```
아래 로직에서 apiResult.result 로 사용을 하시고 있어서 result 의 result 가 조금 어색해보여서요 ㅎㅎ
response로 사용하시는 부분 어떠신가요??